### PR TITLE
SOF-1866: Setup barebone package structure

### PR DESCRIPTION
**Pull request by KvelaGorrrrnio**
The repository is setup for developing a TypesScript application targeting a node (20+) platform. Highlights for the setup:
- TypeScript compiler (`tsc`) is used for building. The TS configuration is nearly identical to sofie-server configuration. The code compiles to a CommonJS setup, while the tools are setup to run as ES modules.
- `ts-node` and `nodemon` are used for setting up the development server.
- Vitest is used as the test runner and assertion library.
- Eslint is used for linting. There are more formatting rules applied here than in the sofie server and sofie web client.
- `yarn npm audit` is used for dependency validation on moderate threads for production dependencies.
- `license-checker` is used for ensuring that we only use packages with useable licenses.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,13 @@
+node_modules/
```
Comment (LindvedKrvang) [inline .gitignore:1]: Would be nice to also add `.idea/` so we ignore JetBrains specific files.
Same for `.vscode`.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
+{
+  "name": "sofie-tv2-server",
+  "version": "0.0.0",
+  "repository": "git@github.com:tv2/sofie-tv2-server.git",
+  "author": "Anders Frederik Jørgensen <afjo@tv2.dk>",
```
Comment (LindvedKrvang) [inline package.json]: Do we want authors?

**Review by LindvedKrvang — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (RasmusAlbrektsen) [inline package.json]: A good author, would recommend.

File: src/name-generator.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export class NameGenerator {
+  public getName(): string {
+    return 'Sofie TV 2 server'
```
Comment (RasmusAlbrektsen) [inline src/name-generator.ts:3]: Should move this string to a constant, this is kind of sloppy Anders. Seriously dude.

**Review by RasmusAlbrektsen — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (KvelaGorrrrnio) [reply .gitignore:1]: That should be in your global gitignore, not in the repository :)

File: package.json (hunk 2 above)
Comment (KvelaGorrrrnio) [reply package.json]: I don't think we want a single author. I have moved it to contributors.

**Pull request merged**